lnrpc/routerrpc/router_backend.go

```diff
@@ -757,6 +758,16 @@ func (r *RouterBackend) extractIntentFromSendRequest(
 			return nil, err
 		}
 
+		// Validate the features if any was specified.
+		if features != nil {
+			err = feature.ValidateDeps(features)
+			if err != nil {
+				return nil, err
```

Comment:
This is an opportunity to start returning better information back to the caller to explain why the payment can't succeed. As a test I tried to pay Conner's AMP invoice from my unupgraded node and got "no routes found" on the command line rather than "payment type unsupported" or something like that. 

In order to do this, we'd need to add the new error return type in path finding itself if we fail to find a path due to a destination node's feature incompatibility. Alternatively, we can do a check for dest feature bit compatibility much earlier in the cycle.

Reply:
I think that would mean it actually did not find a route? Since if it did, the destination node should send an error message because of the missing AMP payload, which would be the error you would see.

Reply:
Yeah so I think we'll end up doing this in the follow up PR that adds the basic invoicing information. In either case, this would've ideally been done in the _prior_ release so older nodes will, know they can't pay an invoice since it's AMP, and not that there doesn't exist a path.